eng/pipelines/templates/stages/vmr-build.yml

```diff
@@ -697,6 +697,18 @@ stages:
             targetOS: osx
             targetArchitecture: x64
 
+        - template: ../jobs/vmr-build.yml
+          parameters:
+            buildName: OSX_Mono
+            isBuiltFromVmr: ${{ parameters.isBuiltFromVmr }}
+            vmrBranch: ${{ variables.VmrBranch }}
+            architecture: x64
+            pool:
+              vmImage: ${{ variables.defaultPoolNameMac }}
+            useMonoRuntime: true
```

Comment:
I'm not 100% sure this does the same as what we currently do for the "Mono runtime pack" case in the runtime official builds since this came from the source-build/s390x use case: it eventually passes `--usemonoruntime` to runtime's build.sh which essentially sets `/p:PrimaryRuntimeFlavor=Mono` but in runtime-official.yml we build `-s mono+libs+host+packs` instead.

The RuntimeFlavor logic in Subsets.props is quite complicated so we might be introducing differences here.